### PR TITLE
Implement SIZEOF and STARTOF for section types

### LINKED ISSUE #1204: [Feature request] `SIZEOF` and `STARTOF` for section types
For example, `SIZEOF(SRAM) == $2000` and `STARTOF(SRAM) == $a000`.

These would have to be link-time values with new RPN opcodes, since `sectionTypeInfo` can be updated by some linker settings.

Alternatively, we could wait for hardware description files to be implemented, and require those to make the final section type metadata available at assembly time.

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #1204

### CODE REVIEW

**Comment by Rangi42** [comment]
pret will be able to get rid of these: https://github.com/pret/pokecrystal/blob/master/constants/hardware_constants.asm#L6-L17

**Comment by ISSOtm** [comment]
What are they useful for, anyway?

File: include/linkdefs.h [hunk 1]
```diff
@@ -56,6 +56,8 @@ enum RPNCommand {
 	RPN_BANK_SELF		= 0x52,
 	RPN_SIZEOF_SECT		= 0x53,
 	RPN_STARTOF_SECT	= 0x54,
+	RPN_SIZEOF_SECTTYPE     = 0x55,
```
Comment (ISSOtm) [inline include/linkdefs.h:59]: Nitpick, but `SEG` is more consistent with... other environments. But maybe we'll update the terminology wholesale when implementing #524 instead?

**Comment by Rangi42** [comment]
There are routines that do things like clearing an entire RAM bank at once.

File: man/rgbds.5 [hunk 2]
```diff
@@ -349,23 +349,33 @@ after it.
...
 .Ar section Ap s Cm STRING
 name.
-.It Li $54 Ta Fn STARTOF section ,
+.It Li $54 Ta Fn STARTOF section ;
 followed by the
 .Ar section Ap s Cm STRING
 name.
+.It Li $55 Ta Fn SIZEOF sectiontype ;
+followed by the
+.Ar sectiontype Ap s Cm LONG
```
Comment (ISSOtm) [inline man/rgbds.5:371]:
```suggestion
.Ar sectiontype Ap s Cm BYTE
```

File: man/rgbds.5 [hunk 3]
```diff
@@ -349,23 +349,33 @@ after it.
...
+value
+.Pq see the Ar Type No values in Sx Sections .
+.It Li $56 Ta Fn STARTOF sectiontype ;
+followed by the
+.Ar sectiontype Ap s Cm LONG
```
Comment (ISSOtm) [inline man/rgbds.5:376]:
```suggestion
.Ar sectiontype Ap s Cm BYTE
```

**Review by ISSOtm — DISMISSED** [review]
Okay, great!

Oops, that was supposed to be requesting changes...

**Comment by ISSOtm** [comment]
> There are routines that do things like clearing an entire RAM bank at once.

I'd argue that much is an antipattern :<

File: include/linkdefs.h (hunk 1 above)
Comment (Rangi42) [reply include/linkdefs.h:59]: "seg" and "segment" are not yet terms that we use anywhere.

**Comment by Rangi42** [comment]
Time travel to 1995 and tell Game Freak. :P

File: include/linkdefs.h (hunk 1 above)
Comment (ISSOtm) [reply include/linkdefs.h:59]: I think it'll be worth switching the terminology then.

Comment (Rangi42) [reply include/linkdefs.h:59]: Can that be a separate issue? "Refer to 'segments' not 'sections' everywhere"?

Comment (ISSOtm) [reply include/linkdefs.h:59]: Yeah, we'll do that when introducing HDFs.

File: man/rgbasm.5 [hunk 4]
```diff
@@ -487,10 +487,12 @@ is in.
 .Ar symbol
 must have been defined already.
 .It Fn SIZEOF arg Ta Returns the size of the section named
```
Comment (ISSOtm) [inline man/rgbasm.5:489]: How about: "If *arg* is a string expression, returns the size of the section named *arg*; if *arg* names a section type, returns the size of that section type."

I think that reads better.

File: man/rgbds.5 [hunk 5]
```diff
@@ -383,12 +393,12 @@ Checks if the value is a valid
 vector, that is one of $00, $08, $10, $18, $20, $28, $30, or $38.
 The value is then ORed with $C7
 .Pq Ql \&| $C7 .
-.It Li $80 Ta Integer literal.
-Followed by the
+.It Li $80 Ta Integer literal;
+followed by the
```
Comment (ISSOtm) [inline man/rgbds.5:397]:
```suggestion
.It Li $80 Ta Integer literal; followed by the
```

File: man/rgbds.5 [hunk 6]
```diff
@@ -383,12 +393,12 @@ Checks if the value is a valid
...
 .Cm LONG
 integer.
-.It Li $81 Ta A symbol's value.
-Followed by the symbol's
+.It Li $81 Ta A symbol's value;
+followed by the symbol's
```
Comment (ISSOtm) [inline man/rgbds.5:401]:
```suggestion
.It Li $81 Ta A symbol's value; followed by the symbol's
```

**Review by ISSOtm — DISMISSED** [review]
Please forgive me for being inflexible with the documentation... :')

File: man/rgbasm.5 (hunk 4 above)
Comment (Rangi42) [reply man/rgbasm.5:489]: Sure. I edited it a bit to match to wording of `BANK` above.

**Review by ISSOtm — APPROVED** [review]
I could nitpick more, but let's not.

**Pull request merged**